pkg/skaffold/deploy/kubectl.go

```diff
@@ -170,24 +174,44 @@ func (k *KubectlDeployer) manifestFiles(manifests []string) ([]string, error) {
 }
 
 // readManifests reads the manifests to deploy/delete.
-func (k *KubectlDeployer) readManifests(ctx context.Context) (deploy.ManifestList, error) {
+func (k *KubectlDeployer) readManifests(ctx context.Context, offline bool) (deploy.ManifestList, error) {
 	// Get file manifests
 	manifests, err := k.Dependencies()
 	if err != nil {
 		return nil, fmt.Errorf("listing manifests: %w", err)
 	}
 
 	// Append URL manifests
+	hasURLManifest := false
 	for _, manifest := range k.KubectlDeploy.Manifests {
 		if util.IsURL(manifest) {
 			manifests = append(manifests, manifest)
+			hasURLManifest = true
 		}
 	}
 
 	if len(manifests) == 0 {
 		return deploy.ManifestList{}, nil
 	}
 
+	// In case no URLs are provided, we can stay offline - no need to run "kubectl create" which would try to connect to a cluster
```

Comment:
Please handle the easy flow early so that the exception here can be de-idented as much as possible.

`if !offline {
  return k.kubectl.ReadManifests(ctx, manifests)
}

...
do other things
`